### ISSUE TITLE
Trades endpoint Internal Server error when omitting counter_asset_type param

### ISSUE THREAD

**Issue by Synesso** [stellar/go#466]
As per https://stellar.stackexchange.com/questions/1098/internal-error-on-trades-api

https://horizon.stellar.org/trades?base_asset_type=native&counter_asset_code=MOBI&counter_asset_issuer=GA6HCMBLTZS5VYYBCATRBRZ3BZJMAFUDKYYF6AH6MVCMGWMRDNSWJPIH returns

```
{
  "type": "https://stellar.org/horizon-errors/server_error",
  "title": "Internal Server Error",
  "status": 500,
  "detail": "An error occurred while processing this request.  This is usually due to a bug within the server software.  Trying this request again may succeed if the bug is transient, otherwise please report this issue to the issue tracker at: https://github.com/stellar/go/services/horizon/internal/issues. Please include this response in your issue."
}
```

User states that the problem is resolved by including the optional parameter `counter_asset_type` .

**Comment by michaelran16** [comment]
According to: https://www.stellar.org/developers/horizon/reference/resources/trade.html
It didn't mention 'counter_asset_type' as optional. 
Could you let me know which doc you are referring to? cc @Synesso

**Comment by Synesso** [comment]
@michaelran16 https://www.stellar.org/developers/horizon/reference/endpoints/trades.html

Optional or not is a side issue. Missing params shouldn't 500.

**Comment by michaelran16** [comment]
@Synesso thanks for the explanation, I will come up with a fix.

**Comment by michaelran16** [comment]
A fix - https://github.com/stellar/go/pull/472

**Comment by howardtw** [comment]
Should this issue have been closed by #474? https://github.com/stellar/go/pull/474#issuecomment-394697003

**Comment by bartekn** [comment]
@howardtw thanks for finding this. It was actually closed by using `action.SetInvalidField` in https://github.com/stellar/go/blame/408036e1b75b0557cfa40a53582128339e19756d/services/horizon/internal/actions_trade.go#L81. Closing.
